internal/errors/errors.go

```diff
@@ -0,0 +1,23 @@
+/*
+Copyright The ORAS Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package errors
+
+import "errors"
+
+// ErrInvalidReference creates a new error based on the reference string.
+func ErrInvalidReference(reference string) error {
```

Comment:
Originally I am thinking the same thing, but looks like this error is most used for processing user input and displaying, I didn't see any use case we wil need to do sth like `err.(type)==ErrInvalidReference`

Reply:
Okay

Reply:
Some udpates on this:
- added `new` prefix to the error generation function
- moved the error to main package since it's directly related to commandline processing/displaying